File: ui-api/src/main/java/io/tipsters/uiapi/Application.java
package io.tipsters.uiapi;

import com.fasterxml.jackson.databind.ObjectMapper;
import com.fasterxml.jackson.databind.SerializationFeature;
import com.fasterxml.jackson.datatype.jsr310.JavaTimeModule;

import org.springframework.boot.SpringApplication;
import org.springframework.boot.autoconfigure.SpringBootApplication;
import org.springframework.context.annotation.Bean;
import org.springframework.http.converter.json.Jackson2ObjectMapperBuilder;

@SpringBootApplication
public class Application {

  @Bean
  public ObjectMapper objectMapper(Jackson2ObjectMapperBuilder objectMapperBuilder) {
    ObjectMapper objectMapper = objectMapperBuilder.build();
    objectMapper.configure(SerializationFeature.WRITE_DATES_AS_TIMESTAMPS, false)
        .registerModule(new JavaTimeModule());
    return objectMapper;
  }

  public static void main(String...args) {
    SpringApplication.run(Application.class, args);
  }
}

File: web-server/src/main/kotlin/io/tipsters/Application.kt
package io.tipsters

import com.fasterxml.jackson.databind.ObjectMapper
import com.fasterxml.jackson.databind.ObjectWriter
import com.fasterxml.jackson.databind.SerializationFeature
import com.fasterxml.jackson.datatype.jsr310.JavaTimeModule
import com.fasterxml.jackson.module.kotlin.KotlinModule
import org.springframework.boot.SpringApplication
import org.springframework.boot.autoconfigure.SpringBootApplication
import org.springframework.context.annotation.Bean
import org.springframework.http.converter.json.Jackson2ObjectMapperBuilder

@SpringBootApplication
open class Application {
    @Bean
    open fun objectMapperBuilder(): Jackson2ObjectMapperBuilder =
            Jackson2ObjectMapperBuilder().modulesToInstall(KotlinModule(), JavaTimeModule())

    @Bean
    open fun objectMapper(jackson2ObjectMapperBuilder: Jackson2ObjectMapperBuilder): ObjectMapper {
        val mapper = jackson2ObjectMapperBuilder.build<ObjectMapper>()
        mapper.configure(SerializationFeature.WRITE_DATES_AS_TIMESTAMPS, false)
        return mapper
    }

    @Bean
    open fun objectWriter(objectMapper: ObjectMapper): ObjectWriter = objectMapper.writer()
}

fun main(args: Array<String>) {
    SpringApplication.run(Application::class.java, *args)
}
